src/utils.py

```diff
@@ -0,0 +1,56 @@
+"""Utilities"""
+import os
+from pathlib import Path
+from typing import Set, Union
+
+import pandas as pd
+import yaml
+
+
+def find_prefixes_in_mapping_set(source_df: pd.DataFrame) -> Set[str]:
+    """Find prefixes in mapping set"""
+    df = source_df.copy()
+    cols_with_prefixes = ['subject_id', 'object_id', 'predicate_id']
+    prefixes = set()
+    for col in cols_with_prefixes:
+        col2 = col.replace('id', 'prefix')
+        df[col2] = df[col].apply(lambda x: x.split(':')[0]
+            if isinstance(x, str) else x)  # handles nan
+        prefixes.update(set(df[col2].to_list()))
+    return prefixes
+
+
+def write_sssom(df: pd.DataFrame, config_path: Union[Path, str], outpath: Union[Path, str]):
+    """Writes a SSSOM file with commented metadata at the top of the file.
+
+    Filters only prefxes in curie_map that exist in the mapping set."""
+    temp_filtered_config_path = str(config_path) + '.tmp'
+    # Load config
+    config = yaml.safe_load(open(config_path, 'r'))
+    # Filter curie_map
+    prefixes: Set[str] = find_prefixes_in_mapping_set(df)
+    config['curie_map'] = {k: v for k, v in config['curie_map'].items() if k in prefixes}
+    # Write
+    with open(temp_filtered_config_path, 'w') as f:
+        yaml.dump(config, f)
+    write_tsv_with_comments(df, temp_filtered_config_path, outpath)
+    os.remove(temp_filtered_config_path)
+
+
+def write_tsv_with_comments(df: pd.DataFrame, comments_file: Union[Path, str], outpath: Union[Path, str]):
```

Comment:
## General: `write_tsv_with_comments()`
@matentzn Let me know what the existing best practice is supposed to be for this.
I have this function (and several others it's used with) in the repos for GARD, [MedGen](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1536735796), ICD11, a TIMS repo, and possibly elsewhere. Maybe there's a function in `sssom-py` that does this like I want? I have a dataframe and a path to a `sssom-metadata.yml`, and I want to write that to a file and include the comment at the top.

Reply:
- [ ] @joeflack4 Nico advised in another PR what to do here; what standard `sssom-py` practices to employ, so I will need to do that here.

Reply:
Done. And in doing so, I also encountered some possible bugs and UX issues, and made:
- https://github.com/mapping-commons/sssom-py/issues/513
- https://github.com/mapping-commons/sssom-py/issues/514
- https://github.com/mapping-commons/sssom-py/issues/515
- https://github.com/mapping-commons/sssom-py/issues/516
- https://github.com/mapping-commons/sssom-py/issues/517

Reply:
This should be ready to be marked resolved by a reviewer.

Nico has examined these 5 issues as well.